chemprop/train/run_training.py

```diff
@@ -330,7 +334,11 @@ def run_training(args: TrainArgs,
                         writer.add_scalar(f'validation_{task_name}_{metric}', val_score, n_iter)
 
             # Save model checkpoint if improved validation score
-            mean_val_score = multitask_mean(val_scores[args.metric], metric=args.metric)
+            mean_val_score = multitask_mean(
+                scores=val_scores[args.metric],
+                metric=args.metric,
+                ignore_nan_metrics=args.ignore_nan_metrics
```

Comment:
I think it might be a little confusing to have the val scores sometimes be NaN and sometimes not during the same run, and it might make it even more complicated to debug an issue in that case. So I would potentially be in favor of keeping the current setup where the user specifies whether to ignore NaN metrics or not. But I'll defer to you and I'm happy to try to implement this if you think it would be useful.